oauth2_http/java/com/google/auth/oauth2/ImpersonatedCredentials.java

```diff
@@ -143,6 +146,47 @@ public static ImpersonatedCredentials create(
         .build();
   }
 
+  /**
+   * @param sourceCredentials the source credential used as to acquire the impersonated credentials
```

Comment:
I think the source credentials can be other types. But it can only be user account credentials or service account credentials when reading from json. I mentioned it in the doc string of the fromJson function.

Reply:
To generate tokens it has to be either a  a Google account or a service account: https://cloud.google.com/iam/docs/creating-short-lived-service-account-credentials#sa-credentials-permissions.

Reply:
Done. Thanks for linking the reference! I didn't know this restriction.